### ISSUE TITLE
Please use a secure PRNG for adding noise to the applications list

### ISSUE THREAD

**Issue by jvoisin** [Exodus-Privacy/exodus-android-app#32]
Hello,

in the [`getReports`]( https://github.com/Exodus-Privacy/exodus-android-app/blob/e69ef1cda67339d9268f42e53b62366486dd4573/app/src/main/java/org/eu/exodus_privacy/exodusprivacy/manager/NetworkManager.java#L237 ) method, the code is using the following snippet:

```java
            // Add some random packages to avoid tracking
            Random rand = new Random(Thread.currentThread().getId());
            int alea = rand.nextInt(120) % 10 + 11;
            for(int i = 0 ; i < alea; i++) {
                int val = rand.nextInt(handles.size());
                packages.add(handles.get(val));
            }
```

This could lead, at some degrees, a rogue server to be able to do educated guess about what applications are installed or not. It would be cool to use [SecureRandom]( https://docs.oracle.com/javase/7/docs/api/java/security/SecureRandom.html ) instead.

This is a super-minor issue, feel free to close it with an `E_DONTCARE` :)

**Comment by U039b** [comment]
Hi @jvoisin 
Even if we use a PNRG, 2 full sync of the list of installed applications will allow any rogue server to remove the noise.

**Comment by jvoisin** [comment]
Sure, but I think that some users will install exodus, run it, glance at the results, be horrified, remove some of their applications and then remove exodus because it fulfilled its purpose.

For this usecase, using a better PRNG will slightly improve privacy.

**Comment by pnu-s** [comment]
Fixed by #90
